### PR TITLE
[#XGB-203] encountered problems

### PR DESCRIPTION
**Pull request by fabianhauser**

### CODE REVIEW

File: documents/final-submission-document/chapters/4_implementation.tex [hunk 1]
```diff
@@ -61,15 +61,87 @@ \subsection{Limitations of \emph{\fullref{sec:requirement-audit-trail}}}
...
+\subsection{Limitations of the Openfire \gls{xmpp} Server}\label{sec:limitations-of-the-openfire-xmpp-server}
+
+As discussed in Section~\ref{sec:development-setup}, the openfire \gls{xmpp} server was used in the development setup. This section details on the encountered limitations while implementing the \gls{xmpp-grid} \gls{broker}.
+
+\subsubsection{WebSocket SASL EXTERNAL Support}
+
+At time of this thesis, openfire does not support SASL EXTERNAL in combination with \gls{xmpp} over WebSockets.
+Therefore, the current implementation of the \gls{xmpp-grid} \gls{broker} was developed with \gls{bosh}, but also support communication over WebSockets thanks to the stanza.io \gls{xmpp} library\footnote{\url{https://github.com/legastero/stanza.io}}.
+
+\subsubsection{Lost Updates}
```
Comment (fabianhauser) [inline documents/final-submission-document/chapters/4_implementation.tex]: I was not sure, if we should also talk about parallel edits here, what do you think @raphiz?

Comment (raphiz) [reply documents/final-submission-document/chapters/4_implementation.tex]: As discussed, parallel edits will be discussed in the concurrency chapter

**Comment by raphiz** [comment]
@fabianhauser I improved the language here and there and refined some parts

**Comment by fabianhauser** [comment]
Thanks @raphiz!

**Pull request merged**